src/runtime_src/core/pcie/driver/linux/xocl/subdev/hwmon_sdm.c

```diff
@@ -1409,14 +1409,13 @@ static void destroy_hwmon_sysfs(struct platform_device *pdev)
 	if (!sdm->supported)
 		return;
 
-	sysfs_remove_group(&pdev->dev.kobj, &hwmon_sdm_bdinfo_attrgroup);
-
 	if (sdm->hwmon_dev) {
-		hwmon_device_unregister(sdm->hwmon_dev);
 		device_remove_file(sdm->hwmon_dev, &name_attr.dev_attr);
+		hwmon_device_unregister(sdm->hwmon_dev);
```

Comment:
Raj, could you please explain why we need to change the sequence back from unregister first and remove file second? All linux code is doing this sequence: https://elixir.bootlin.com/linux/latest/source/drivers/hwmon/asc7621.c#L1112

I am wondering if we change it back, we might see the occasionally memory corruption again.

I actually tested all the sequence, so if we just do this sequence, we can see the painc oops right a way.
create the file
register
unregister
remove file << panic here

If we swap the unregister and remove file. The panic seems gone. But why we see the panic is my question.

Reply:
@rajkumar-xilinx I think I kind of understand your are using this in a different way of creating devs, you have 2 devs.... please make sure backport this to 2022.2 after this merged.